spec/views/post/post_index_spec.rb

```diff
@@ -0,0 +1,56 @@
+require 'rails_helper'
+
+RSpec.feature 'Post index page', type: :feature do
+  describe 'index page' do
+    let!(:user) do
+      User.create(name: 'Alex', photo: 'alx.jpg', bio: 'Chemist', postcounter: 5)
+    end
+
+    let!(:lily) do
+      User.create(name: 'lily', photo: 'l.jpg', bio: 'teacher from mexico', postcounter: 0)
+    end
+
+    let!(:post) do
+      Post.create(title: 'Post Title', text: 'Post Text', author: user)
+    end
+
+    let!(:comment) do
+      Comment.create(author: lily, post:, text: 'Great post!')
+    end
+
+    before do
+      visit user_posts_path(user)
+    end
+    it ' can see the user profile picture.' do
+      expect(page).to have_css("img[src*='#{user.photo}']")
+    end
+    it 'I can see the user username.' do
+      expect(page).to have_content(user.name)
+    end
+    it 'show the number of posts the user has written.' do
+      expect(page).to have_content("Number of posts : #{user.postcounter}")
+    end
+
+    it 'I can see a post title.' do
+      expect(page).to have_content(post.title)
+    end
+    it 'displays some of the post\'s body' do
+      expect(page).to have_selector('p', text: 'Post Text')
+    end
+    it 'I can see the first comments on a post.' do
+      post.comments.first(5).each do |comment|
+        expect(page).to have_content(comment.text)
+      end
```

Comment:
- You have done well so far, but I suggest that you use the method, `recent_comments` in the posts model that returns the first 5 comments for a post. Please fix this 😄

Reply:
Hello @mckent05, Thank you for the review it was really helpful 💯 
I already implemented all the required changes.